Harmony/Public/Patching/HarmonyManipulator.cs

```diff
@@ -528,6 +533,28 @@ bool WriteFinalizerCalls(bool suppressExceptions)
 				if (modifiesControlFlow)
 					lastInstruction.OpCode = OpCodes.Ret;
 
+				foreach (var ilman in ilmanipulators)
+				{
+					var methodParamsTypes = ilman.method.GetParameters().Select(x => x.ParameterType).ToArray();
+
+					var contextIndex = Array.IndexOf(methodParamsTypes, typeof(ILContext));
+					var baseIndex = Array.IndexOf(methodParamsTypes, typeof(MethodBase));
```

Comment:
This needs a proper for-loop for checking args and detecting any extra ones (i.e. wrong type or wrong number of params). This is especially important in case we'd like to add more args to manipulators in the future.

For a good example, check out [how arguments array is built for transpilers](https://github.com/BepInEx/HarmonyX/blob/1606fcb27b81e0f0cc869ecb0eeff503bc10fa54/Harmony/Internal/Patching/ILManipulator.cs#L182).